iaso/api/accounts.py

```diff
@@ -57,23 +57,30 @@ class AccountViewSet(ModelViewSet):
     PUT /api/account/<id>
     """
 
-    permission_classes = [
-        permissions.IsAuthenticated,
-        HasPermission(permission.SOURCES),  # type: ignore
-        HasAccountPermission,
-    ]
     serializer_class = AccountSerializer
     results_key = "accounts"
     queryset = Account.objects.all()
     # FIXME: USe a PATCH in the future, it make more sense regarding HTTP method semantic
     http_method_names = ["patch", "put"]
 
+    def get_permissions(self):
+        if self.action == "switch":
+            permission_classes = [permissions.IsAuthenticated, HasAccountPermission]
+        else:
+            permission_classes = [
+                permissions.IsAuthenticated,
+                HasPermission(permission.SOURCES),  # type: ignore
+                HasAccountPermission,
+            ]
+
+        return [permission() for permission in permission_classes]
+
     @action(detail=False, methods=["patch"], url_path="switch")
     def switch(self, request):
         # TODO: Make sure the account_id is present
         self.permission_classes = [permissions.IsAuthenticated, HasAccountPermission]
         self.check_permissions(request)
-        account_id = request.data.get("account_id", None)
+        account_id = int(request.data.get("account_id", None))
```

Comment:
```suggestion
        account_id = int(request.data["account_id"]) if request.data.get("account_id") else None
```

This could be safer because `int(None)` would raise an error:

```python
>>> int(None)
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```